infra/docker/jobservice/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM python:3.7-slim-buster
+FROM jupyter/pyspark-notebook:ae5f7e104dd5
```

Comment:
Hmm this is not ideal, that image is a bit of a kitchen sink (and more than a GB).

But this does raise an interesting question, for best local experience we want to bundle job service with spark, but for production use we could use a much more minimal image. Thoughts (@woop ?)

(not opposed to merging this as is but we'd need to come up with a better way later)

For context, I'm not very concerned about the image _size_ per se, but those jupyter images have a bunch of stuff installed in them in different ways (conda/pip/apt) so it may make this setup more fragile than it needs to.

Reply:
I agree on your points... I spent some time on this and couldn't easily get both proper python and spark versions working properly with each other. Since we already had this image for jupyter notebook I just reused it, but we should definitely have a better base for this... Not sure it's very high priority to figure it out right now though.

Reply:
@tsotnet I think this is the type of tech debt that slowly erodes the product experience. Aren't we already installing this 1+GB image for the Jupyter image? I would love if we could make this one more slim. I am also worried that we will keep delaying cleaning this up until its merged in, so I am more in favour of building our own light weight image.

Reply:
I don't remember anymore, just some runtime or build time errors when installing things. Nothing that can't be resolved though, just a time sink.

Reply:
@woop created an issue here https://github.com/feast-dev/feast/issues/1117